### PR TITLE
feat: process proxy events

### LINKED ISSUE #264: Pass upload events to queue
Let's pass artifacts upload events to event queue. 
Size of what file should I pass? Here is the example. 
image
Also, note, that there can several distribution packages: war AND jar AND jar with dependencies.

### PR DESCRIPTION
**Pull request by olenagerasimova**
closes #264 

Note, that maven artifact consists of [several files](https://repo1.maven.org/maven2/org/bouncycastle/bcpg-jdk15on/1.70/) and minimal distribution contains only pom (as our parent pom project, no jars) file. Thus, we add events to the proxy queue when some archive or pom file is uploaded and it's possible that the same artifact will be added twice to the queue, because of it, we remove the same items from queue when artifact was processed.

### CODE REVIEW

File: src/main/java/com/artipie/maven/metadata/ArtifactEventInfo.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     * @param keys Package item names
+     * @return Key to artifact package
+     */
+    public Key artifactPackage(final Collection<Key> keys) {
+        Key result = keys.stream().findFirst().orElseThrow(
+            () -> new ArtipieException("No artifact files found")
+        );
+        for (final String ext : MavenSlice.EXT) {
+            final Optional<Key> artifact = keys.stream().filter(
+                key -> key.string().endsWith(ext) && !key.string().contains("javadoc")
```
Comment (dgarus) [inline src/main/java/com/artipie/maven/metadata/ArtifactEventInfo.java:34]: `key.string()` is expensive, we should call it only once.

File: src/main/java/com/artipie/maven/http/MavenSlice.java [hunk 2]
```diff
@@ -32,20 +36,36 @@
...
+        )
+    );
 
     /**
      * Ctor.
      * @param storage The storage and default parameters for free access.
      */
     public MavenSlice(final Storage storage) {
-        this(storage, Policy.FREE, Authentication.ANONYMOUS, "*", new EventQueue<>());
+        this(storage, Policy.FREE, Authentication.ANONYMOUS, "*", Optional.of(new LinkedList<>()));
```
Comment (dgarus) [inline src/main/java/com/artipie/maven/http/MavenSlice.java:68]: `Optional.empty()` looks like a better choice.

**Review by dgarus — CHANGES_REQUESTED** [review]
Thanks!
There are a few minor comments.

**Comment by olenagerasimova** [comment]
thanks, corrected

**Review by dgarus — APPROVED** [review]
lgtm

**Pull request merged**